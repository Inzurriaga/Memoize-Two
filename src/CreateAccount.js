import React, { Component } from "react"

class Player{
    constructor(name, password, avatar, ){
        this.name = name;
        this.password = password;
        this.completed = []
    }
}

export default class CreateAccount extends Component{
    constructor(){
        super();
        this.state = {
            username: "",
            password: "",
        }
    }

    username = e => {
        this.setState({
            username: e.target.value
        })
    }
    
    password = e => {
        this.setState({
            password: e.target.value
        })
    }

    switchToLogin = () => {
        this.props.toggle()
    }

    createAccount = e => {
        e.preventDefault()
        // add descructuring here
        let player = new Player(this.state.username, this.state.password)
        this.props.createNewAccount(player)
    }


    render(){
        return(
            <form className="createaccount-section" onSubmit={this.createAccount}>
            <input type="button" onClick={this.switchToLogin} value="login"></input>
                <h2>CREATE ACCOUNT</h2>
                <input onChange={this.username} type="text" placeholder="Username"></input>
                <input onChange={this.password}  type="password" placeholder="Password"></input>
                <input type="submit" value="Create"></input>
            </form>
        )
    }
}