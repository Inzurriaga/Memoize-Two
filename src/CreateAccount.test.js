import React from 'react';
import { shallow } from 'enzyme';
import CreateAccount from "./CreateAccount"

const mockCreateNewAccout = jest.fn()
const mockUsers = [{"name":"duy","password":"duy","completed":[1,2,5]},{"name":"gabriel","password":"gabriel","completed":[1,2,5]},{"name":"isaac","password":"isaac","completed":[1,2]}]
const mockToggleLoginOption = jest.fn()

describe("CreateAccount", ()=> {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <CreateAccount toggle={mockToggleLoginOption}
                      createNewAccount={mockCreateNewAccout} 
                      users={mockUsers}/>
    )
  })

  it("should match Snap shot", () => {
    expect(wrapper).toMatchSnapshot()
  })

  it("should have a default state", () => {
    expect(wrapper.state()).toEqual({
      username: "",
      password: "",
    })
  })

  it("should change the username state when text is type into the username input", () => {
    expect(wrapper.state("username")).toEqual("")
    wrapper.find("input").at(1).simulate("change", {target: {value: "gabriel"}})
    expect(wrapper.state("username")).toEqual("gabriel")
  })

  it("should change the password state when text is type into the password input", () => {
    expect(wrapper.state("password")).toEqual("")
    wrapper.find("input").at(2).simulate("change", {target: {value: "gabriel"}})
    expect(wrapper.state("password")).toEqual("gabriel")
  })

  it("should invoke the toggle function when button is click", () => {
    wrapper.find("input").at(0).simulate("click")
    expect(mockToggleLoginOption).toHaveBeenCalled()
  })

  it("should invoke the create account when button is click", () => {
    wrapper.find("form").simulate("submit", {preventDefault: () => {}})
    expect(mockCreateNewAccout).toHaveBeenCalled()
  })
})