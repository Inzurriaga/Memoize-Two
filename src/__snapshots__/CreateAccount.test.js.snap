// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateAccount should match Snap shot 1`] = `
<form
  className="createaccount-section"
  onSubmit={[Function]}
>
  <input
    onClick={[Function]}
    type="button"
    value="login"
  />
  <h2>
    CREATE ACCOUNT
  </h2>
  <input
    onChange={[Function]}
    placeholder="Username"
    type="text"
  />
  <input
    onChange={[Function]}
    placeholder="Password"
    type="password"
  />
  <input
    type="submit"
    value="Create"
  />
</form>
`;
